### ISSUE TITLE
Diff command is showing differences even when "terraform show" does not.

### ISSUE THREAD

**Issue by azamora-zsc** [infracost/infracost#913]
I find the tool really useful, but this situation is really weird and brings a lot of doubts about it. The matter is that I'm saving some terraform plans from several terraform modules and analyzing them with infracost. 

When I execute terraform show I get the following:
```
$ terraform show  plan.bin 

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:

Terraform will perform the following actions:

Plan: 0 to add, 0 to change, 0 to destroy.
```

Then I save the plan as json by executing this:
`$ terraform show --json plan.bin > plan.json`

and finally this to get the following result:
```
$ infracost diff --path plan.json                                                
Detected Terraform plan JSON file at plan.json

✔ Calculating monthly cost estimate

Project: plan.json

- aws_autoscaling_group.this
  -$12.83

    - aws_launch_configuration.this

        - Instance usage (Linux/UNIX, on-demand, t2.micro)
          -$9.78

        - EC2 detailed monitoring
          -$2.10

        - root_block_device

            - Storage (general purpose SSD, gp2)
              -$0.95

- aws_db_instance.this
  -$130

    - Database instance
      -$124

    - Database storage
      -$5.96

    - Database storage IOPS
      $0.00

- aws_route53_record.cert_validation
  Monthly cost depends on usage

    - Standard queries (first 1B)
      Monthly cost depends on usage
        -$0.40 per 1M queries

    - Latency based routing queries (first 1B)
      Monthly cost depends on usage
        -$0.60 per 1M queries

    - Geo DNS queries (first 1B)
      Monthly cost depends on usage
        -$0.70 per 1M queries

+ aws_route53_record.cert_validation["xx.yy.com"]
  Monthly cost depends on usage

    + Standard queries (first 1B)
      Monthly cost depends on usage
        +$0.40 per 1M queries

    + Latency based routing queries (first 1B)
      Monthly cost depends on usage
        +$0.60 per 1M queries

    + Geo DNS queries (first 1B)
      Monthly cost depends on usage
        +$0.70 per 1M queries

- aws_route53_record.guac_cdn_cert_validation
  Monthly cost depends on usage

    - Standard queries (first 1B)
      Monthly cost depends on usage
        -$0.40 per 1M queries

    - Latency based routing queries (first 1B)
      Monthly cost depends on usage
        -$0.60 per 1M queries

    - Geo DNS queries (first 1B)
      Monthly cost depends on usage
        -$0.70 per 1M queries

+ aws_route53_record.guac_cdn_cert_validation["xx.zz.com"]
  Monthly cost depends on usage

    + Standard queries (first 1B)
      Monthly cost depends on usage
        +$0.40 per 1M queries

    + Latency based routing queries (first 1B)
      Monthly cost depends on usage
        +$0.60 per 1M queries

    + Geo DNS queries (first 1B)
      Monthly cost depends on usage
        +$0.70 per 1M queries

+ module.bastion.aws_autoscaling_group.this[0]
  +$12.83

    + module.bastion.aws_launch_configuration.this[0]

        + Instance usage (Linux/UNIX, on-demand, t2.micro)
          +$9.78

        + EC2 detailed monitoring
          +$2.10

        + root_block_device

            + Storage (general purpose SSD, gp2)
              +$0.95

+ module.db-replica.module.db_instance.aws_db_instance.this[0]
  +$130

    + Database instance
      +$124

    + Database storage
      +$5.96

    + Database storage IOPS
      $0.00

+ module.db.module.db_instance.aws_db_instance.this[0]
  +$130

    + Database instance
      +$124

    + Database storage
      +$5.96

    + Database storage IOPS
      $0.00

Monthly cost change for plan.json
Amount:  +$130 ($144 -> $274)
Percent: +90%

----------------------------------
Key: ~ changed, + added, - removed

To estimate usage-based resources use --usage-file, see https://infracost.io/usage-file
```

Why is this happening? Is this a bug or something?

**Comment by aliscott** [comment]
Thanks @azamora-zsc, yeah this looks like a bug with parsing some resources in modules.

Are you able to provide the following?
1. An example of the Terraform HCL code so we can reproduce
2. Terraform version
3. Infracost version

**Comment by azamora-zsc** [comment]
Hi @aliscott, thank you for your quick answer. The versions used are:
Infracost v0.9.5
Terraform v0.12.31

As you can see the Terraform version is pretty old, and that could be a problem. It should be updated in the near future.
About the HLC code, I cannot provide the original code... Let's see if I can reproduce the bug with some other code and post it here.

**Comment by aliscott** [comment]
Thanks @azamora-zsc!

**Comment by angelichorsey** [comment]
I'm also encountering this issue, except I do have changes. It notices those, but also has a bunch of +/- from resources starting with `module.` and also some of the sub resources. Seems to somewhat cancel out...but I haven't debugged it fully as I was just playing with `infracost` for the first time today.

Infracost v0.9.7
Terraform v0.12.31
+ provider.aws v3.59.0
+ provider.null v3.1.0
+ provider.template v2.2.0

I'll come back to this if I find anything.

**Comment by aliscott** [comment]
Thanks @angelichorsey. If you have any example terraform code that can reproduce this then please add it here.

**Comment by aliscott** [comment]
https://github.com/infracost/infracost/pull/968 might fix part of this issue. This fixes an issue with any resources that have a `.` in the index part, e.g. `aws_instance.my_instance["index.1"]`.

**Comment by aliscott** [comment]
I've dug into this more and I'm able to reproduce the issues with Terraform v0.12.31. So will work on a fix and keep this issue updated.

**Comment by aliscott** [comment]
@azamora-zsc and @angelichorsey thanks for reporting this bug! We've released a fix for this in v0.9.8, so please let us know if this helps.
